### PR TITLE
Improved REPL

### PR DESCRIPTION
**Pull request by finnbar**
REPL has been improved to make it more useful. Changes are as follows:

* Added file loading via `:c`, which previously did nothing. (The `compile` function returned unit before.)
* Allow definitions to be written on the REPL for later use, via `let x = <expr>` just like in an actual program. Expressions are also allowed, and are still prefaced by an implicit `let it =`.

### CODE REVIEW

**Review by mbg — COMMENTED** [review]
LGTM, but probably will need rebasing after #2 is merged as well.

**Comment by finnbar** [comment]
Just realised that this branch has an old Parser.hs on it, since I didn't regenerate it with happy after changing the parser. Will delete the Lexer.hs and Parser.hs files after #2 is merged, which solves this problem.

**Comment by finnbar** [comment]
Okay, this has been brought up to date with master despite my best attempts - so should be ready to merge.

File: src/Interpreter.lhs [hunk 1]
```diff
@@ -129,7 +130,13 @@
...
     
 >   eval :: GlEnv -> Expr -> Either String Expr
->   eval env e = evalState (runErrorT (eval' e)) env
+>   eval env e = evalState (runExceptT (eval' e)) env
+
+>   evalDef :: GlEnv -> Definition -> Either String GlEnv
+>   evalDef env (Def var e) = do
+>       ty <- infer e (toTyEnv env)
+>       let gldef = GlDef {glExpr = e, glType = ty}
+>       return $ (var, gldef) : env
```
Comment (mbg) [inline src/Interpreter.lhs]: I am wondering if there is a better name for this function than `evalDef`, given that it doesn't seem to be doing any evaluation. How about something like `addDef`? Indeed, that already exists in https://github.com/mbg/simply-typed-ext/blob/3b91d874260a87169a43265489e17af3ba532e67/app/Main.lhs#L35

**Review by mbg — APPROVED** [review]

**Pull request merged**